### PR TITLE
Do migration

### PR DESCRIPTION
**Pull request by AndyBoWu**
poc to launch a node in DO

### CODE REVIEW

File: tf/do/main.tf [hunk 1]
```diff
@@ -0,0 +1,220 @@
+provider "digitalocean" {
+    token = var.do_token
+}
+
+resource "random_id" "droplet_id" {
+    byte_length = 8
```
Comment (LeoHChen) [inline tf/do/main.tf]: I actually want to name the droplet based on uptimerobot rule to make it easier to identify.

File: tf/do/main.tf [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+resource "digitalocean_droplet" "harmony_node" {
+    name        = "node-s${var.shard}-${var.blskey_index}-${random_id.droplet_id.hex}"
+    image       = var.droplet_system_image
+    region      = var.droplet_region
+    size        = var.droplet_size
+    tags        = ["harmony"]
+    # ssh_keys    = [digitalocean_ssh_key.harmony_ssh_key.fingerprint]
+    ssh_keys = [25624276]
```
Comment (LeoHChen) [inline tf/do/main.tf]: what's this?

File: tf/do/main.tf [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+resource "digitalocean_firewall" "harmony_fw" {
+    name = "harmony-fw"
+
+    droplet_ids = [digitalocean_droplet.harmony_node.id]
+
+    inbound_rule {
+        protocol         = "tcp"
+        port_range       = "22"
+        source_addresses = ["0.0.0.0/0", "::/0"]
```
Comment (LeoHChen) [inline tf/do/main.tf]: please don't open it general public. SSH has to be whitelisted.

File: tf/do/main.tf [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    inbound_rule {
+        protocol         = "tcp"
+        port_range       = "9500"
+        source_addresses = ["0.0.0.0/0", "::/0"]
+    }
+}
+
+# test on Andy's local env
```
Comment (LeoHChen) [inline tf/do/main.tf]: don't submit any test code in PR.

File: tf/do/r.list [hunk 5]
```diff
@@ -0,0 +1,9 @@
+nyc1	UP
```
Comment (LeoHChen) [inline tf/do/r.list]: is this all DO region?

File: tf/do/variables.tf [hunk 6]
```diff
@@ -0,0 +1,855 @@
+variable "do_token" {
+  description = "Your DigitalOcean API token"
+  default     = "da616614ccb17b31ef8c0cf993f36885ea084001d626926e1a4ea75f7da68d66"
```
Comment (LeoHChen) [inline tf/do/variables.tf]: man, this is huge security issue! never expose any token or key to github PR.  Please change your token on DO, and only save it locally, and use some environment variable or cli parameter to pass in to the recipe.

File: tf/do/variables.tf [hunk 7]
```diff
@@ -0,0 +1,855 @@
...
+}
+
+variable "droplet_size" {
+  description = "The size of the Droplet"
+  default     = "s-2vcpu-2gb"
+}
+
+variable "droplet_system_image" {
+  description = "The system image of the Droplet"
+  default     = "centos-6-x64"
```
Comment (LeoHChen) [inline tf/do/variables.tf]: use centos-8

Comment (AndyBoWu) [reply tf/do/variables.tf]: updated to `centos-8-x64`

File: tf/do/variables.tf (hunk 6 above)
Comment (AndyBoWu) [reply tf/do/variables.tf]: for testing only, it has been deleted.

File: tf/do/main.tf (hunk 2 above)
Comment (AndyBoWu) [reply tf/do/main.tf]: This has been updated to use a regular ssh key pair.

File: tf/do/files/fast.sh [hunk 8]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (AndyBoWu) [reply tf/do/files/fast.sh]: deleted fast.sh

File: tf/do/main.tf (hunk 3 above)
Comment (AndyBoWu) [reply tf/do/main.tf]: fixed

File: tf/do/r.list (hunk 5 above)
Comment (AndyBoWu) [reply tf/do/r.list]: yes, multiple regions supported

**Pull request merged**